src/main/java/com/apicatalog/rdf/impl/RdfNQuadImpl.java

```diff
@@ -37,10 +38,39 @@ public Optional<RdfResource> getGraphName() {
 
     @Override
     public String toString() {
-        return  "RdfNQuadImpl[subject=" + getSubject()
-                    + ", predicate=" + getPredicate()
-                    + ", object=" + getObject()
-                    + ", graph=" + graphName
-                    + "]";
+        final StringBuilder builder = new StringBuilder()
+                .append(getSubject())
+                .append(' ')
+                .append(getPredicate())
+                .append(' ')
+                .append(getObject())
+                .append(' ');
+
+        if (graphName != null) {
+            builder.append(graphName).append(' ');
+        }
+
+        return builder.append('.').toString();
+    }
+
+    @Override
+    public int hashCode() {
+        final int prime = 31;
+        int result = super.hashCode();
+        result = prime * result + Objects.hash(graphName);
+        return result;
     }
+
+    @Override
+    public boolean equals(Object obj) {
+        if (this == obj)
+            return true;
+        if (!super.equals(obj))
+            return false;
+        if (getClass() != obj.getClass())
```

Comment:
Shouldn't this check instanceOf RdfNquad instead, to allow for other implementations?

Reply:
you are right, class comparison does not allow to mix different RDF implementations. (FYI I let Eclipse generate those methods) Please note the other RDF primitives `equals` implementation.
The question is, do we need to support different RDF primitives implementations now? If yes, then please feel free to raise an issue, as it requires changes to all the primitives, not just to the `RdfNQuadsImpl`. 

The other thing is, that the future of RDF primitives/implementation, that is part of Titanium, is not quite clear right now, I would like to separate it from the project. 

Reply:
@hmottestad Just using this opportunity, you have mentioned a couple of times that you are a leading developer in RDF4J. I'm looking for a simple and viable way to enable seamless and vital integration between Titanium and other libraries.  Perhaps, this could be interesting for you https://github.com/apache/jena/discussions/2216

Reply:
I have a fork of Titanium where I've modified the code to use generics. Also some changes to support a statement collector instead of returning an RDF dataset.

It's far from backwards compatible, but it makes it possible to allow for very little conversion.